### PR TITLE
Submit api testing

### PR DESCRIPTION
**Pull request by newhoggy**
Integration test for `cardano-submit-api`.  This is run in CI.

### CODE REVIEW

File: cardano-testnet/src/Testnet/SubmitApi.hs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import           Text.Show
+
+import qualified Data.Aeson as J
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.ByteString as BSS
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.ByteString.Char8 as BS8
+import qualified Data.HashMap.Lazy as HM
+import qualified Data.List as L
+import qualified Data.Text as T
```
Comment (erikd) [inline cardano-testnet/src/Testnet/SubmitApi.hs:47]: There are only 26 letters in the alphabet, so using single letters for qualified imports may mean we run out of letters. I would suggest `List` and `Text` for these.

File: cardano-testnet/src/Testnet/SubmitApi.hs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import qualified Data.Text.Encoding as T
+import qualified Data.Time.Clock as DTC
```
Comment (erikd) [inline cardano-testnet/src/Testnet/SubmitApi.hs:49]: And `Time` for this.

File: cardano-testnet/src/Testnet/SubmitApi.hs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import qualified Hedgehog as H
+import qualified Hedgehog.Extras.Stock.IO.File as IO
+import qualified Hedgehog.Extras.Stock.IO.Network.Socket as IO
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Hedgehog.Extras.Stock.String as S
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.Concurrent as H
+import qualified Hedgehog.Extras.Test.File as H
+import qualified Hedgehog.Extras.Test.Network as H
+import qualified Hedgehog.Extras.Test.Process as H
```
Comment (erikd) [inline cardano-testnet/src/Testnet/SubmitApi.hs:60]: Is there any name collision when importing all `Hedgehog.Extra` modules as `HExtra` ?

**Review by erikd — APPROVED** [review]
Not sure about how you are naming qualified imports, but otherwise :+1:

File: cardano-testnet/src/Testnet/SubmitApi.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-testnet/src/Testnet/SubmitApi.hs:47]: Fixed!

File: cardano-testnet/src/Testnet/SubmitApi.hs (hunk 3 above)
Comment (newhoggy) [reply cardano-testnet/src/Testnet/SubmitApi.hs:60]: No know name collisions as yet.

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**